tensorflow_probability/python/mcmc/internal/util.py

```diff
@@ -728,3 +728,37 @@ def index_remapping_gather(params,
     return dist_util.move_dimension(result_t,
                                     source_idx=broadcast_indices_ndims - 1,
                                     dest_idx=axis)
+
+
+# TODO(b/111801087): Use a standardized API, when available.
+@make_innermost_getter
+def get_field(kernel_results, field_name):
+  """Get field value from kernel_results or kernel_results.accepted_results."""
+  attr = getattr(kernel_results, field_name, None)
+  if attr is not None:
+    return attr
+  accepted_results = getattr(kernel_results, 'accepted_results', None)
+  if accepted_results is None:
+    raise TypeError('Cannot extract {} from {}'.format(
+        field_name, kernel_results))
+  attr = getattr(accepted_results, field_name, None)
+  if attr is None:
+    raise TypeError('Cannot extract {} from {}'.format(
+        field_name, kernel_results))
```

Comment:
This should probably be generalized to `get_field(accepted_results, field_name)`, to support MH(Something(HMC)).  We don't currently have any of the latter, but it shouldn't hurt.  (Unless I'm missing something and that doesn't actually work?)